Update README.md (#42)
nanoframework/nanoFramework.M5Stack: README.md

## Usage

Those 4 nugets provide a support for the M5 Stack's Core (gray) [M5Stack](https://docs.m5stack.com/en/products?id=core), [M5StickC](https://docs.m5stack.com/en/core/m5stickc), [M5StickCPlus](https://docs.m5stack.com/en/core/m5stickc_plus) and [M5Core2](https://docs.m5stack.com/en/core/core2).

IMPORTANT: The M5Stack NuGet Package must only be used for application targeting the M5 Stack's Core (gray); the other 3 NuGets must be used individually depending on your target according its name.
NOTE1: Before trying to add NuGet packages to your projects and/or before flashing the devices (see next section) using MS Visual Studio (VS), open VS > Tools > Options > NuGet Package Manager > Package Sources  and make sure that it contains an entry pointing to https://api.nuget.org/v3/index.json , otherwise add it.

> NOTE2: When invoking VS > Project > Manage NuGet Packages ... make sure that the "include prerelease" checkbox is clicked/selected and that in the Package source drop-down menu (right upper corner) "nuget.org" is selected.  

Reviewer: This is required if you're targeting preview versions, which may not be the case... Your struggle with this came from the fact that there was no stable version released when you started working on this.
```suggestion
NOTE2: When invoking VS > Project > Manage NuGet Packages make sure that in the Package source drop-down menu (right upper corner) "nuget.org" is selected. Also if you're using preview version the "include prerelease" checkbox should be clicked/selected as well.  
```

The NuGets bring support for the screens as well and require to be flashed with the proper image (using dotnet CLI):

```shell

> # Replace the com port number by your COM port (on WIndows check it with the Device Manager) 

Reviewer: ```suggestion
# Replace `COM3` with the appropriate number of the COM port to which your device is connected. (on Windows you can check this in the Device Manager).
```

# For the M5Core2:
nanoff --target M5Core2 --update --preview --serialport COM3
```

> NOTE3: If the nanoff commands fail, make sure you did what is written on NOTE1.

Reviewer: ```suggestion
NOTE3: If the nanoff commands fails, make sure you have followed instruction from NOTE1 above.
```

Once you have the nugets, you can then enjoy accessing the screen, the accelerometer, get a Grove I2C connecter, add events on the buttons. And you don't even need to think about anything, all is done for you in the most transparent way!

> Note: All the classes that you'll have access are all using the Lazy pattern to be instantiated including the screen. This have the advantage to use as little memory and setup time as possible.

In the samples below, we'll use either M5Stack, either M5Stick as examples, they are all working in a very similar way.

### namespaces

Make sure you add the proper namespaces reference to your C# program header e.g.
using nanoFramework;

### Screen

> The only thing you need to do to access the screen is to initialize it; nevetheless this is also target specific e.g.:

Reviewer: ```suggestion
The only thing you need to do to access the screen is to initialize it (please note that `InitializeScreen()` it's platform specific) e.g.:
```

Author: in this one i don't actually agree, it is not a platform, it is actually an specific target within a family of products , maybe of the same platform....if you call platform to all M5 Stack (company) products and e..g another platform the STM32 family of products....again, here it is target specific , each M5Stack target needs a different screen initialization semantics ...(which is not nice actually)...

Reviewer: Yes, that should be target, not platform.

```csharp

> For Core use: M5Stack.InitializeScreen();

Reviewer: These code blocks are meant to be copy/paste, so the code inside has to be valid.
If you want to add examples for Core and StickCPlus you have to have to 2 code blocks with the note before each one explaining on which target it belongs to.

Author: ok , i understand...but i would suggest that you can implement that small change as you did with the text ,...otherwise the overhead is too high...and i dont actually understand yet how should i proceed now...on Github i have two options, either "Commit suggestion" or "Resolve Conversation"...what does the later do??

Reviewer: @jcabad100 check [here](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/commenting-on-a-pull-request#resolving-conversations). 
You should mark a conversation as resolved if you're addressing directly the comment with a change that matches the suggestion of the reviewer. If you replying back or adding new ideas/suggestions, you should leave it open. Then it could be up to you or the reviewer to close it.

As for the overhead you mention, not sure exactly what you mean... if this about the changes, know that you can make small changes directly on github or on an open PR.

Screen.Write(0, 0, 10, 10, toSend);
```

> The Console class works in a similar way as the classic `System.Console` i.e. it writes to the remote Debbuger output on the host (VS):

Reviewer: No need to bring in Debug.WriteLine to the conversation and the extra explanation... that makes it more confusing...
```suggestion
The Console class works in a similar way as the classic `System.Console`. In order to use it you have to reference it using the fully qualified name of the methods, like this:
```

> 

Reviewer: ```suggestion

```

> e.g. Debug.WriteLine("Hello from nanoFramework 1!");

Reviewer: ```suggestion
```

> and Console.Write("Hello from nanoFramework 2!"); will both write on the Debugger output , not on the target device screen; in order to do that

Reviewer: ```suggestion
```

> you have to use the nanoFramework.Console instead e.g.

Reviewer: ```suggestion
```

};
```

> > Note: The M5Core2 has touch screen and the buttons are "virtual"". At the time of this writting the .NET nanoFramework distributed team is already implementing and testing the touch-screen (touch-panel) functionality (will be released soon).  

Reviewer: ```suggestion
> Note: The M5Core2 has touch screen and the buttons are "virtual"". At the time of this writing the .NET nanoFramework team is implementing and testing the touch-screen (touch-panel) functionality (will be released soon).
```
